File: 백준/Silver/10845. 큐/큐.cc
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    queue<int> que;
    int n, val;
    string inst;
    cin >> n;
    while (n--) {
        cin >> inst;
        if (inst == "push") {
            cin >> val;
            que.emplace(val);
        } else if (inst == "pop") {
            if (que.size()) {
                cout << que.front() << '\n';
                que.pop();
            } else {
                cout << -1 << '\n';
            }
        } else if (inst == "size") {
            cout << que.size() << '\n';
        } else if (inst == "empty") {
            cout << que.empty() << '\n';
        } else if (inst == "front") {
            if (que.size()) {
                cout << que.front() << '\n';
            } else {
                cout << -1 << '\n';
            }
        } else {
            if (que.size()) {
                cout << que.back() << '\n';
            } else {
                cout << -1 << '\n';
            }
        }
    }
}

File: 백준/Silver/10845. 큐/큐.java
import java.util.*;
import java.io.*;

public class Main {
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        Deque<Integer> queue = new ArrayDeque<>();
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String command = st.nextToken();

            switch (command) {
                case "push":
                    queue.offer(Integer.parseInt(st.nextToken()));
                    break;
                case "pop":
                    if (queue.isEmpty()) sb.append(-1).append('\n');
                    else sb.append(queue.poll()).append('\n');
                    break;
                case "size":
                    sb.append(queue.size()).append('\n');
                    break;
                case "empty":
                    sb.append(queue.isEmpty() ? 1 : 0).append('\n');
                    break;
                case "front":
                    if (queue.isEmpty()) sb.append(-1).append('\n');
                    else sb.append(queue.peek()).append('\n');
                    break;
                case "back":
                    if (queue.isEmpty()) sb.append(-1).append('\n');
                    else sb.append(queue.peekLast()).append('\n');
                    break;
            }
        }

        System.out.print(sb);
    }
}
